src/WebJobs.Script/Utility.cs

```diff
@@ -411,11 +411,14 @@ internal static void AddFunctionError(IDictionary<string, ICollection<string>> f
             functionName = isFunctionShortName ? functionName : Utility.GetFunctionShortName(functionName);
 
             ICollection<string> functionErrorCollection = new Collection<string>();
-            if (!functionErrors.TryGetValue(functionName, out functionErrorCollection))
+            if (!string.IsNullOrEmpty(functionName))
```

Comment:
This fixes #3581

Reply:
nit: reduce nesting here by adding the check to the same `if` statement below.